cfme/utils/appliance/__init__.py

```diff
@@ -2137,45 +2146,116 @@ def check_domain_enabled(self, domain):
-            'cp {template_dir}/etc/httpd/conf.d/manageiq-external-auth.conf.erb {manageiq_ext_auth}'
-            .format(template_dir=template_dir,
-                    manageiq_ext_auth=manageiq_ext_auth))
-        self.ssh_client.run_command('setenforce 0 && '
-                                    'systemctl restart sssd && '
-                                    'systemctl restart httpd')
+            'cp {t}{c}.erb {c}'
+            .format(t=template_dir, c=self.CONF_FILES['httpd_ext_auth'])
+        )
+        assert self.ssh_client.run_command('setenforce 0')
+        self.sssd.restart()
+        self.httpd.restart()
+        self.wait_for_web_ui()
+
+        # UI configuration of auth provider type
+        self.server.authentication.configure_auth(auth_mode='external', **kwargs)
+
+    @logger_wrap('Disabling openldap external auth provider')
+    def disable_openldap(self):
+        self.server.authentication.configure_auth()
+        files_to_remove = [
+            '/etc/sssd/sssd.conf',
+            '/etc/pam.d/httpd-auth',
+            self.HTTPD_CONF_MODULES['manageiq_ext_auth'],
+            self.HTTPD_CONF_MODULES['manageiq_remoteuser']
+        ]
+        for file in files_to_remove:
+            assert self.ssh_client.run_command('rm -rf {}'.format(file))
+        self.evmserverd.restart()
+        self.httpd.restart()
         self.wait_for_web_ui()
+        self.server.authentication.configure_auth(auth_mode='database')
+
+    @logger_wrap('Configuring freeipa external auth provider')
+    def configure_freeipa(self, ipaserver, principal, password, **kwargs):
+        """Configure appliance UI and backend for freeIPA
+
+        Args:
+            ipaserver: string server hostname/ip
+            principal: string ipa principal
+            password: string ipa password
+        Notes:
+            Also takes ipadomain and iparealm for backend config.
+            Passes kwarg to UI configuration call
+        """
+        if self.is_pod:
+            # appliance_console_cli fails when calls hostnamectl --host. it seems docker issue
+            # raise BZ ?
+            assert str(self.ssh_client.run_command('hostname')).rstrip() == self.fqdn
+
+        # First, clear any existing ipa config, runs clean if not configured
+        assert self.ssh_client.run_command('appliance_console_cli --uninstall-ipa')
+        self.wait_for_web_ui()  # httpd restart in uninstall-ipa
+
+        # ext auth ipa requires NTP sync
+        if ipaserver not in self.server.settings.ntp_servers_values:
+            self.server.settings.update_ntp_servers({'ntp_server_1': ipaserver})
+            # sleep(120)
+
+        # backend appliance configuration of ext auth provider
+        self.appliance_console_cli.configure_ipa(
+            ipaserver, principal, password,
+            kwargs.pop('ipadomain') if 'ipadomain' in kwargs else '',  # pop, not in ui form
+            kwargs.pop('iparealm') if 'iparealm' in kwargs else ''
```

Comment:
**QUESTION**
Do you really need to pop here? Otherwise, you can do like this:
```python
kwargs.get('ipadomain', ''),
kwargs.get('iparealm', '')
```

Reply:
I used pop here because the ipadomain/iparealm aren't used in `authentication.configure_auth(auth_mode='external', **kwargs)` below.

This definitely has room for improvement, and I plan on that in #6538. These methods should use the new auth provider objects (#6612), which provide `as_fill_value()` and `as_external_value()` methods.  I split some of these changes into separate PRs for easier initial reviews. Would you rather see some different syntax here in the meantime?